app/commands/user/reputation_token/award_for_pull_requests.rb

```diff
@@ -0,0 +1,38 @@
+class User
+  class ReputationToken
+    class AwardForPullRequests
+      include Mandate
+
+      initialize_with :user
+
+      def call
+        return if user.github_username.empty?
+
+        pull_requests.find_each do |pr|
+          User::ReputationToken::AwardForPullRequest.(pr.data[:action], pr.data[:author], pr.data)
+        rescue StandardError => e
+          Rails.logger.error "Error syncing pull request reputation for user #{user.handle}: #{e}"
+        end
+      end
+
+      private
+      def pull_requests
+        authored_pull_requests.or(merged_pull_requests).or(reviewed_pull_requests)
+      end
+
+      def authored_pull_requests
+        ::Git::PullRequest.left_joins(:reviews).where(author_github_username: user.github_username)
```

Comment:
Why `left join` not `inner join`?

Reply:
Well, it should not be an inner join here as we're only concerned about the pull requests, and not about the reviews. That said, the `def reviewed_pull_requests` method could use an inner join. I tried to do that with `joins` instead of `left_joins`, but that I get:

```
ArgumentError: Relation passed to #or must be structurally compatible. Incompatible values: [:joins]
```

which is due to me doing: `authored_pull_requests.or(merged_pull_requests).or(reviewed_pull_requests)`. I could split this up into separate calls if you'd prefer that.